nemo/collections/nlp/data/dialogue_state_tracking_sgd/data_processor.py

```diff
@@ -0,0 +1,676 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+# Copyright 2019 The Google Research Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This file contains code artifacts adapted from the original implementation:
+https://github.com/google-research/google-research/blob/master/schema_guided_dst/baseline/data_utils.py
+"""
+
+import collections
+import json
+import os
+import pickle
+import re
+from typing import Dict, List
+
+import numpy as np
+
+from nemo.collections.nlp.data.dialogue_state_tracking_sgd.input_example import InputExample
+from nemo.utils import logging
+
+__all__ = ['SGDDataProcessor']
+
+FILE_RANGES = {
+    "sgd_single_domain": {"train": range(1, 44), "dev": range(1, 8), "test": range(1, 12)},
+    "sgd_multi_domain": {"train": range(44, 128), "dev": range(8, 21), "test": range(12, 35)},
+    "sgd_all": {"train": range(1, 128), "dev": range(1, 21), "test": range(1, 35)},
+    "sgd_all_single": {"train": range(1, 128), "dev": range(1, 8), "test": range(1, 12)},
+    "multiwoz": {"train": range(1, 18), "dev": range(1, 3), "test": range(1, 3)},
+    "debug_sample": {"train": range(1, 2), "dev": range(1, 2), "test": range(1, 2)},
+}
+
```

Comment:
Only [sgd_all, sgd_all_single, sgd_debug] are in the config, others are not supported?

Reply:
For the others you need to change the dataset and is not for release